docs/markdown/snippets/dependency_include_dirs.md

```diff
@@ -0,0 +1,8 @@
+## All dependencies now separate include directory and compile args
+
+Prior to 1.2.0, dependencies declared with `declare_dependency` would separate
+-I/-isystem into separate fields, but external dependencies would combine them
+as raw arguments. This meant that using `dependency.partial_dependency(includes
+: true)` would not work as expected with external dependencies, and made the
+implementation more complicated. As of 1.2.0 this has been changed to separate
```

Comment:
I don't think we need to mention the implementation complexity in the release notes.

We should just highlight the fact that this change is about making the behavior consistent, so that `dependency()` and "dependency via subproject fallback" don't do different things.